scheduler/scheduler_medium_test.go

```diff
@@ -372,6 +377,35 @@ func TestStopTask(t *testing.T) {
 			So(tsk.State(), ShouldEqual, core.TaskStopped)
 		})
 	})
+
+	Convey("Calling StopTask on a firing task", t, func() {
+		Convey("Should allow last scheduled workfow execution to finish", func() {
+			c.timeToWait = 500 * time.Millisecond
+			lse := fixtures.NewListenToSchedulerEvent()
+			s.eventManager.RegisterHandler("Scheduler.TaskStopped", lse)
+			sc := schedule.NewWindowedSchedule(time.Second, nil, nil, 0)
+			t, _ := s.CreateTask(sc, w, false)
+			startTime := time.Now()
+			t.(*task).Spin()
+			//allowing things to settle and waiting for task state to change to firing
+			time.Sleep(100 * time.Millisecond)
+			So(t.State(), ShouldResemble, core.TaskFiring)
+			// stop task when task state is firing
+			t.(*task).Stop()
+			// the last scheduled workflow execution should be allowed to finish
+			// so we expect that stopping the task is going to happen not early than 500ms (set by by timeToWait)
+			//the task should have fired once
+			So(t.HitCount(), ShouldEqual, 1)
```

Comment:
Please move checking the hitCount into the `case <-lse.TaskStoppedEvents` after the checking that task is Stopped (so the HitCount for 100% sure does not change)